docs/proposals/implemented/2017-08-split-plans-from-serviceclass.md

```diff
@@ -0,0 +1,124 @@
+# Split Plans from ServiceClasses
+
+Proposed: 2017-08-08
+Approved: 2017-08-15
+
+## Abstract
+
+Proposal to split Plans from ServiceClass to become a top-level object in our API.
+
+## Motivation
+
+If we get a ServiceClass, we get all of the Plans, which potentially contain a
+lot of data. The data of concern is primarily the Schema information used for
+exposing the details of a ServicePlan to a UI.
+
+## Constraints and Assumptions
+
+ - It is desired to have the ServicePlan data in one place and not
+   duplicated. Duplicated data gets out of sync.
+
+## Proposed Design
+
+ - new ServicePlan resource based on existing ServicePlan object
+ - Implement FieldSelector for ServicePlan on `Free` field combined with ServiceClass Reference
+
+### ServiceClass API Changes
+
+Instead of having an array of ServicePlans defined inline, we have an array of
+Reference to ServicePlan.
+
+``` go
+// ServiceClass represents an offering in the service catalog.
+type ServiceClass struct {
+    ...
+
+    // All references to ServicePlans are removed
+    
+    ...
+}
+```
+
+### ServicePlan API Resource
+
+Added the kubernetes TypeMeta and ObjectMeta.
+
+Annotations for client and not being namespaced.
+
+No other changes.
+
+```go
+// ServicePlan represents a tier of a ServiceClass.
+type ServicePlan struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    ...
+    
+        // ServiceClass is the Class of Service that this plan implements. 
+	ServiceClass v1.LocalObjectReference `json:"serviceClass"`
+}
+```
+
+```
+// ServicePlanList is a list of instances.
+type ServicePlanList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []ServicePlan `json:"items"`
+}
+```
+
+Classes are not namespaced, so plans are not namespaced. Possibility
```

Comment:
nit: such resources are called "cluster scoped" (as opposed to "namespace scoped") in Kubernetes

Reply:
okay, will edit.